blog/models.py

```diff
@@ -11,7 +11,7 @@ class Post(models.Model):
     published_date = models.DateTimeField(blank=True, null=True)
 
     def publish(self):
-        self.published_date = timezone.now()
+        self.published_date = timezone.now
```

Comment:
여기에는 `timezone.now()`가 맞는 것 같은데 바꾸신 이유가 있을까요? 🤔 
=> 보통 이렇게 변경하는 경우 연관된 테스트가 터져야 테스트가 제 기능을 하고 있는 것이라.. 기존 테스트가 없다면 하나 새로 작성해주셔도 좋을 것 같습니다.
```python
In [29]: from django.utils import timezone

In [30]: timezone.now
Out[30]: <function django.utils.timezone.now()>

In [31]: timezone.now()
Out[31]: datetime.datetime(2021, 8, 2, 4, 46, 48, 80042, tzinfo=<UTC>)
```

Reply:
blog.Post.created_date: (fields.W161) Fixed default value provided.
	HINT: It seems you set a fixed date / time / datetime value as default for this field. This may not be what you want. If you want to have the current date as default, use `django.utils.timezone.now`

기본적으로 함수에 ()를 붙여서 사용해서 습관처럼 변경했었는데 변경하자마자 test코드를 실행할 때, 해당하는 메세지가 보여서 괄호를 제거하는 것이 default 사용이라고 생각했습니다...

Reply:
`self.published_date = timezone.now()` 이렇게 테스트 돌리셨는데 위 메시지 출력됐다는 말씀이시죠?
 혹시 어떻게 돌리셨는지 알 수 있을까요? 제 로컬에서 재연 좀 해보려고 합니다.

Reply:
```
# blog/models.py Line 10
created_date = models.DateTimeField(default=timezone.now())
```

위처럼 작성했을 때 메세지가 출력됐습니다!

Reply:
아하 저는 publish method 관련 코멘트였습니다. 저처럼 한번 shell에서 수행해보시겠어요?
```python
In [1]: Post.objects.all()[0]
Out[1]: <Post: >

In [2]: post = _

In [3]: post.publish()
<function now at 0x102be1790> <class 'function'>

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-6673668f8b81> in <module>
----> 1 post.publish()

~/workspace/my-first-blog/blog/models.py in publish(self)
     14         self.published_date = timezone.now
     15         print(self.published_date, type(self.published_date))
---> 16         self.save()
     17 
     18     def approved_comments(self):

~/.pyenv/versions/3.9.4/envs/rookie/lib/python3.9/site-packages/django/db/models/base.py in save(self, force_insert, force_update, using, update_fields)
    726                 update_fields = frozenset(loaded_fields)
    727 
--> 728         self.save_base(using=using, force_insert=force_insert,
    729                        force_update=force_update, update_fields=update_fields)
    730     save.alters_data = True

~/.pyenv/versions/3.9.4/envs/rookie/lib/python3.9/site-packages/django/db/models/base.py in save_base(self, raw, force_insert, force_update, using, update_fields)
    757             if not raw:
    758                 self._save_parents(cls, using, update_fields)
--> 759             updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)
    760         # Store the database on which the object was saved
    761         self._state.db = using

~/.pyenv/versions/3.9.4/envs/rookie/lib/python3.9/site-packages/django/db/models/base.py in _save_table(self, raw, cls, force_insert, force_update, using, update_fields)
    820                       for f in non_pks]
    821             forced_update = update_fields or force_update
--> 822             updated = self._do_update(base_qs, using, pk_val, values, update_fields,
    823                                       forced_update)
    824             if force_update and not updated:

~/.pyenv/versions/3.9.4/envs/rookie/lib/python3.9/site-packages/django/db/models/base.py in _do_update(self, base_qs, using, pk_val, values, update_fields, forced_update)
    870             else:
    871                 return False
--> 872         return filtered._update(values) > 0
    873 
    874     def _do_insert(self, manager, using, fields, update_pk, raw):

~/.pyenv/versions/3.9.4/envs/rookie/lib/python3.9/site-packages/django/db/models/query.py in _update(self, values)
    710         query.add_update_fields(values)
    711         self._result_cache = None
--> 712         return query.get_compiler(self.db).execute_sql(CURSOR)
    713     _update.alters_data = True
    714     _update.queryset_only = False

~/.pyenv/versions/3.9.4/envs/rookie/lib/python3.9/site-packages/django/db/models/sql/compiler.py in execute_sql(self, result_type)
   1377         related queries are not available.
   1378         """
-> 1379         cursor = super().execute_sql(result_type)
   1380         try:
   1381             rows = cursor.rowcount if cursor else 0

~/.pyenv/versions/3.9.4/envs/rookie/lib/python3.9/site-packages/django/db/models/sql/compiler.py in execute_sql(self, result_type, chunked_fetch, chunk_size)
   1053             result_type = NO_RESULTS
   1054         try:
-> 1055             sql, params = self.as_sql()
   1056             if not sql:
   1057                 raise EmptyResultSet

~/.pyenv/versions/3.9.4/envs/rookie/lib/python3.9/site-packages/django/db/models/sql/compiler.py in as_sql(self)
   1343                     )
   1344             else:
-> 1345                 val = field.get_db_prep_save(val, connection=self.connection)
   1346 
   1347             # Getting the placeholder for the field.

~/.pyenv/versions/3.9.4/envs/rookie/lib/python3.9/site-packages/django/db/models/fields/__init__.py in get_db_prep_save(self, value, connection)
    765     def get_db_prep_save(self, value, connection):
    766         """Return field's value prepared for saving into a database."""
--> 767         return self.get_db_prep_value(value, connection=connection, prepared=False)
    768 
    769     def has_default(self):

~/.pyenv/versions/3.9.4/envs/rookie/lib/python3.9/site-packages/django/db/models/fields/__init__.py in get_db_prep_value(self, value, connection, prepared)
   1429         # Casts datetimes into the format expected by the backend
   1430         if not prepared:
-> 1431             value = self.get_prep_value(value)
   1432         return connection.ops.adapt_datetimefield_value(value)
   1433 

~/.pyenv/versions/3.9.4/envs/rookie/lib/python3.9/site-packages/django/db/models/fields/__init__.py in get_prep_value(self, value)
   1408 
   1409     def get_prep_value(self, value):
-> 1410         value = super().get_prep_value(value)
   1411         value = self.to_python(value)
   1412         if value is not None and settings.USE_TZ and timezone.is_naive(value):

~/.pyenv/versions/3.9.4/envs/rookie/lib/python3.9/site-packages/django/db/models/fields/__init__.py in get_prep_value(self, value)
   1269     def get_prep_value(self, value):
   1270         value = super().get_prep_value(value)
-> 1271         return self.to_python(value)
   1272 
   1273     def get_db_prep_value(self, value, connection, prepared=False):

~/.pyenv/versions/3.9.4/envs/rookie/lib/python3.9/site-packages/django/db/models/fields/__init__.py in to_python(self, value)
   1369 
   1370         try:
-> 1371             parsed = parse_datetime(value)
   1372             if parsed is not None:
   1373                 return parsed

~/.pyenv/versions/3.9.4/envs/rookie/lib/python3.9/site-packages/django/utils/dateparse.py in parse_datetime(value)
    105     Return None if the input isn't well formatted.
    106     """
--> 107     match = datetime_re.match(value)
    108     if match:
    109         kw = match.groupdict()

TypeError: expected string or bytes-like object

```

Reply:
이건 shell_plus 설치 후 publish 직접한번 호출해보는걸로 하시죠

Reply:
0d1b94e 추가했습니다!

Reply:
간단하게만 확인했는데 일단 커밋 하나에 너무 많은 변경사항이 들어있는 것 같아요. 다음부터는 쪼개주시면 좋을 것 같습니다.

Reply:
추가로 publish 사용하는 테스트도 하나 있으면 좋겠습니다.
만약 테스트가 있었으면 timezone.now로 변경했을 때 테스트가 터져서 어느 부분이 잘못됐는지 쉽게 파악이 가능했을거에요.

Reply:
b92c6b1 추가했습니다!